src/Compilers/Core/Portable/InternalUtilities/EnumField.cs

```diff
@@ -71,10 +71,10 @@ private class EnumFieldComparer : IComparer<EnumField>
             int IComparer<EnumField>.Compare(EnumField field1, EnumField field2)
             {
                 // Sort order is descending value, then ascending name.
-                var diff = unchecked((long)field2.Value - (long)field1.Value);
+                int diff = unchecked(((long)field2.Value).CompareTo((long)field1.Value));
```

Comment:
Do we still need the `unchecked`?

Reply:
I believe the conversion from `ulong` to `long` requires it, per Section 6.2.1, "Explicit numeric conversions":

> In a checked context, the conversion succeeds if the value of the source operand is within the range of the destination type, but throws a System.OverflowException if the value of the source operand is outside the range of the destination type.

Reply:
I'm terrified that the compiler devs aren't sure.

Reply:
We compile all our code in unchecked though ...

Reply:
I have confirmed that it is required if we ever compile in a checked context. (which I thought we did for Debug?) Anyway, I dislike ever relying on ambient checked/unchecked context since it has no way to demonstrate the intent of the developer.